src/toggle/Toggle.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { pure, skinnable, props, t } from '../utils';
+import cx from 'classnames';
+import { getValueLink } from '../link-state';
+
+
+/**
+ * A nice animated Toggle rendered using only CSS
+ */
+@pure
+@skinnable()
+@props({
+  /** The current value (`true` if checked) */
+  value: t.maybe(t.Boolean),
+  /** Callback called when user clicks on the Toggle */
+  onChange: t.maybe(t.Function),
+  /** To be used together with `linkState` */
+  valueLink: t.maybe(t.struct({
+    value: t.Boolean,
+    requestChange: t.Function
+  })),
+  /**
+   * The size for the Toggle in whatever unit (px, em, rem ...).
+   * It will be used to compute `width`, `height` and `border-radius` as follows:
+   * `width: size`, `height: size / 2`, `border-radius: size / 2`
+   */
+  size: t.maybe(t.union([t.String, t.Number])),
+  className: t.maybe(t.String),
+  style: t.maybe(t.Object)
+})
+export default class Toggle extends React.Component {
+
+  componentDidMount() {
+    this.updateCheckbox(this.props);
+  }
+
+  updateCheckbox = (props) => {
+    const { value } = getValueLink(this, props);
+    const { checkbox } = this.refs;
+    const checkboxNode = checkbox.nodeType === 1 ?
+      checkbox :
+      checkbox.getDOMNode();
+    checkboxNode.checked = value;
+  }
+
+  getHalfSize(size) {
+    if (t.String.is(size)) {
+      const numberMatch = (/\d+/).exec(size);
+      const unitMatch = (/[a-z]+/).exec(size);
+
+      const number = numberMatch ? parseInt(numberMatch[0], 10) : '';
+      const unit = unitMatch ? unitMatch[0] : '';
+      return `${number / 2}${unit}`;
```

Comment:
This is scary and not safe for corner cases like '13.5px' or 'px'. No easy way to get the computed size instead? Otherwise you can rely on the scary but spec'd feature of `parseInt/parseFloat` of ignoring non-digits in a string when parsing